qa/rpc-tests/wallet_mergetoaddress_2.py

```diff
@@ -0,0 +1,341 @@
+
+    def z_mergetoaddress_tz2t(self, utxos_quantity, notes_quantity):
+        mytaddr2 = self.nodes[2].listaddresses()[0]
+        listunspent_before_merge = self.nodes[2].listunspent(0)
+        balance_before_merge = self.nodes[2].z_getbalance(mytaddr2)
+        result = self.nodes[0].z_mergetoaddress(["*"], mytaddr2, 0, utxos_quantity, notes_quantity)        
+        wait_and_assert_operationid_status(self.nodes[0], result['opid'], "success", "", 3600)
+        self.sync_all()
+        self.nodes[1].generate(1)
+        self.sync_all()
+        tx = self.get_merged_transaction(listunspent_before_merge, self.nodes[2].listunspent(0))
+        return {self.op_result_k: result, self.balance_delta_k: self.nodes[2].z_getbalance(mytaddr2) - balance_before_merge, self.joinsplit_len_k: len(tx["vjoinsplit"])}
+
+    def z_mergetoaddress_tz2z(self, utxos_quantity, notes_quantity):
+        if (len(self.nodes[2].z_listaddresses()) == 0):
+            myzaddr2 = self.nodes[2].z_getnewaddress()
+        else:
+            myzaddr2 = self.nodes[2].z_listaddresses()[0]
+        listunspent_before_merge = self.nodes[2].z_listunspent(0)
+        balance_before_merge = self.nodes[2].z_getbalance(myzaddr2)
+        result = self.nodes[0].z_mergetoaddress(["*"], myzaddr2, 0, utxos_quantity, notes_quantity)        
+        wait_and_assert_operationid_status(self.nodes[0], result['opid'], "success", "", 3600)
+        self.sync_all()
+        self.nodes[1].generate(1)
+        self.sync_all()
+        tx = self.get_merged_transaction(listunspent_before_merge, self.nodes[2].z_listunspent(0))
+        return {self.op_result_k: result, self.balance_delta_k: self.nodes[2].z_getbalance(myzaddr2) - balance_before_merge, self.joinsplit_len_k: len(tx["vjoinsplit"])}
+
+    def setup_chain(self):
+        self.import_data_to_data_dir()
+        print("Initializing test directory "+self.options.tmpdir)
+        initialize_chain_clean(self.options.tmpdir, 3)
+
+    def setup_network(self, split=False):
+        args = ['-debug=zrpcunsafe', '-experimentalfeatures', '-zmergetoaddress', '-maxtipage=36000000']
+        self.nodes = []
+        self.nodes.append(start_node(0, self.options.tmpdir, args))
+        self.nodes.append(start_node(1, self.options.tmpdir, args))
+        args2 = ['-debug=zrpcunsafe', '-experimentalfeatures', '-zmergetoaddress', '-mempooltxinputlimit=7', "-maxtipage=36000000"]
+        self.nodes.append(start_node(2, self.options.tmpdir, args2))
+        connect_nodes_bi(self.nodes,0,1)
+        connect_nodes_bi(self.nodes,1,2)
+        connect_nodes_bi(self.nodes,0,2)
+        self.is_network_split=False
+        self.sync_all()
+
+    def run_test (self):
+
+        # Node0 acts as "from" (merge_to_address sender)
+        # Node1 acts as miner and funds distributor
+        # Node2 acts ad "to" (merge_to_address receiver)
+
+        #(generate 1b on n0, generate 100bs on node1, shield 1 b on n0) * 100 + generate 1b on n1
```

Comment:
Does this line refer to the state imported with `import_data_to_data_dir`? If so, could you please make it explicit?

Reply:
Yes, it is related to the state imported with `import_data_to_data_dir`, added explicit comment on that function.